src/twisted/web/http.py

```diff
@@ -974,12 +974,13 @@ def getHeader(self, key):
         """
         Get an HTTP request header.
 
-        @type key: C{bytes}
```

Comment:
Is it necessary to duplicate interfaces' docstrings on their implementing classes when they're identical? Current practice is mixed, and it seems redundant, since the generated documentation will come out the same either way (the implementing class's methods will be documented with text taken from the interface).

Reply:
(I should add that that comment isn't intended as an objection to this PR.)

Reply:
I know that I've seen pydoctor inherit the docstrings before, so I removed them dupes from this PR, but it doesn't seem to be working properly in the case of `http.Request` or `server.Request`:

![image](https://user-images.githubusercontent.com/43662/55286043-31858f80-534b-11e9-8659-91130292c832.png)

![image](https://user-images.githubusercontent.com/43662/55286039-23377380-534b-11e9-8100-768a922f0ea9.png)

pydoctor is especially confused in the latter case &mdash; it lists the `Componentized` methods several times.

Reply:
(note: pydoctor's confusion there is fixed on current pydoctor, as seen on 18.9s docs)

Reply:
Yes, the `Componentized` bit is fixed with pydoctor trunk.

It still doesn't inherit the documentation, though. `method`, `uri`, `path`, and `args` aren't shown, and the methods show up as undocumented:

![image](https://user-images.githubusercontent.com/43662/55293059-b18d1300-53a6-11e9-939d-73f0f168ac55.png)

Reply:
Maybe this is happening because of the mix of inheritance and interface annotation here?

```
http.Request      implements    IConsumer
     |            defines       setResponseCode
     |                          setHeader
     |
server.Request    implements    IRequest
```

In any case, since `http.Request` doesn't implement `IRequest` it needs to document these methods and attributes on its own. (Indeed, there are other subclasses of `http.Request` that don't implement `IRequest` either — `ProxyRequest` and `ReverseProxyRequest`).